aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java

```diff
@@ -85,70 +84,32 @@ private AppSyncRequestFactory() {}
      * @throws DataStoreException On Failure to inspect
      */
     @NonNull
-    static <T extends Model> String buildSyncDoc(
-            @NonNull final Class<T> modelClass,
+    static <T, M extends Model> AppSyncGraphQLRequest<T> buildSyncRequest(
+            @NonNull final Class<M> modelClass,
             @Nullable final Long lastSync,
             @SuppressWarnings("SameParameterValue") @Nullable final String nextToken)
             throws DataStoreException {
 
-        final StringBuilder doc = new StringBuilder();
-
-        ModelSchema schema;
         try {
-            schema = ModelSchema.fromModelClass(modelClass);
-        } catch (AmplifyException amplifyException) {
-            throw new DataStoreException("Failed to get fields for model.",
-                    amplifyException, "Validate your model file.");
-        }
-        final String capitalizedPluralName = Casing.capitalizeFirst(schema.getPluralName());
-
-        int indent = 0;
-        // Outer container, e.g. query SyncBlogPost {
-        doc.append("query Sync").append(capitalizedPluralName).append(" {\n");
-
-        // Inner directive, e.g. syncBlogPosts(
-        doc.append(padBy(++indent)).append("sync").append(capitalizedPluralName);
+            AppSyncGraphQLRequest.Builder builder = AppSyncGraphQLRequest.builder()
+                    .modelClass(modelClass)
+                    .operation(QueryType.SYNC)
+                    .requestOptions(new DataStoreGraphQLRequestOptions())
+                    .responseType(new TypeToken<Iterable<String>>(){}.getType());
```

Comment:
You have your little type factory gizmo where you can do like `TypeFactory.of(Iterable.class, String.class)` right? That'd remove the `gson` import from the top of the doc, which would be cool from an encapsulation standpoint.

Reply:
Updated!